### DISCUSSION TITLE
Does date-fns work on Nginx servers?

### DISCUSSION THREAD

**Discussion by Dhaigh94** [js-fns/date-fns#3032] [General] [answered]
I am having issues with date-fns where I deploy my app to a Nginx server and it is sending the correct dates but the next day it is sending the incorrect dates unless I redeploy my code. Is there a way for date-fns to grab the current date and time when the function is called while it is on a server?

My code example is
```
const { add, sub, formatISO } = require('date-fns');
const { utcToZonedTime, format } = require('date-fns-tz');

const date = new Date();
const timeZone = 'Australia/Sydney';
const zonedDate = utcToZonedTime(date, timeZone);

function todaysDate() {
	const pattern = 'dd-MM-yyyy';
	return format(zonedDate, pattern);
}
```

**Comment by fturmel** [comment, marked as answer]
Hi @Dhaigh94,

I think this is a simple scope issue in your code. The `date` and `zonedDate` constants are outside of the function body, so every subsequent call will use the same date.

Try something like this instead:

```js
const { utcToZonedTime, format } = require('date-fns-tz');

const timeZone = 'Australia/Sydney';
const pattern = 'dd-MM-yyyy';

function todaysDate() {
  const zonedDate = utcToZonedTime(new Date(), timeZone);
  return format(zonedDate, pattern);
}
```
